openedx/core/djangoapps/course_groups/views.py

```diff
@@ -145,40 +153,80 @@ def course_cohort_settings_handler(request, course_key_string):
-                'cohorted_inline_discussions', divided_inline_discussions
+                'divided_inline_discussions', divided_inline_discussions
             )
             settings_to_change['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
 
-        if 'always_cohort_inline_discussions' in request.json:
+        if 'always_divide_inline_discussions' in request.json:
             settings_to_change['always_divide_inline_discussions'] = request.json.get(
-                'always_cohort_inline_discussions'
+                'always_divide_inline_discussions'
             )
 
         if not settings_to_change:
             return JsonResponse({"error": unicode("Bad Request")}, 400)
 
         try:
+            if settings_to_change:
+                discussion_settings = set_course_discussion_settings(course_key, **settings_to_change)
+
+        except ValueError as err:
+            # Note: error message not translated because it is not exposed to the user (UI prevents this state).
+            return JsonResponse({"error": unicode(err)}, 400)
+
+    return JsonResponse(_get_course_discussion_settings_representation(
+        course,
+        discussion_settings
+    ))
+
+
+@require_http_methods(("GET", "PATCH"))
+@ensure_csrf_cookie
+@expect_json
+@login_required
+def course_cohort_settings_handler(request, course_key_string):
+    """
+    The restful handler for cohort setting requests. Requires JSON.
+    This will raise 404 if user is not staff.
+    GET
+        Returns the JSON representation of cohort settings for the course.
+    PATCH
+        Updates the cohort settings for the course. Returns the JSON representation of updated settings.
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    # Although this course data is not used this method will return 404 is user is not staff
+    course = get_course_with_access(request.user, 'staff', course_key)
+
+    if request.method == 'PATCH':
+        settings_to_change = {}
+
+        if 'is_cohorted' in request.json:
+            settings_to_change['is_cohorted'] = request.json.get('is_cohorted')
+
+        if not settings_to_change:
+            return JsonResponse({"error": unicode("Bad Request")}, 400)
+
+        try:
+            is_cohorted = None
             if 'is_cohorted' in settings_to_change:
                 is_cohorted = settings_to_change['is_cohorted']
                 cohorts.set_course_cohorted(course_key, is_cohorted)
-                del settings_to_change['is_cohorted']
-                settings_to_change['division_scheme'] = CourseDiscussionSettings.COHORT if is_cohorted \
-                    else CourseDiscussionSettings.NONE
-            if settings_to_change:
-                discussion_settings = set_course_discussion_settings(course_key, **settings_to_change)
+
+            scheme = CourseDiscussionSettings.COHORT if is_cohorted else CourseDiscussionSettings.NONE
```

Comment:
I was imagining this could all be simplified to remove settings_to_change, since is_cohorted is the only thing that can be in it. How about this?

````
if request.method == 'PATCH':
    if not 'is_cohorted' in request.json:
        return JsonResponse({"error": unicode("Bad Request")}, 400)
    is_cohorted = request.json.get('is_cohorted')
    try:
        cohorts.set_course_cohorted(course_key, is_cohorted)
        scheme = CourseDiscussionSettings.COHORT if is_cohorted else CourseDiscussionSettings.NONE
        scheme_settings = {'division_scheme': scheme}
        set_course_discussion_settings(course_key, **scheme_settings)
    except ValueError as err:
         ...